handler/core/strategy/hmacsha.go

```diff
@@ -0,0 +1,36 @@
+package strategy
+
+import (
+	"github.com/ory-am/fosite"
+	"github.com/ory-am/fosite/enigma"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+type HMACSHAStrategy struct {
+	Enigma enigma.HMACSHAEnigma
+}
+
+func (h HMACSHAStrategy) GenerateAccessToken(_ context.Context, _ *http.Request, requester fosite.AccessRequester, _ interface{}) (token string, signature string, err error) {
+	return h.Enigma.Generate(requester.GetClient().GetHashedSecret())
```

Comment:
Yep! I've looked through the code and all seems well. We can now adapt the stateless behavior as we want :)